topdown/aggregates.go

```diff
@@ -11,16 +11,18 @@ import (
 	"github.com/open-policy-agent/opa/topdown/builtins"
 )
 
-func builtinCount(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+func builtinCount(bctx BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
 	switch a := operands[0].Value.(type) {
-	case *ast.Array:
-		return iter(ast.IntNumberTerm(a.Len()))
-	case ast.Object:
-		return iter(ast.IntNumberTerm(a.Len()))
-	case ast.Set:
+	case interface{ Len() int }:
 		return iter(ast.IntNumberTerm(a.Len()))
 	case ast.String:
 		return iter(ast.IntNumberTerm(len([]rune(a))))
+	case ast.Ref:
+		rterm := bctx.e.generateVar("res")
+		return bctx.e.unify(ast.NewTerm(a), rterm, func() error {
+			plugged := bctx.e.bindings.Plug(rterm)
+			return builtinCount(bctx, []*ast.Term{plugged}, iter)
+		})
```

Comment:
ℹ️ Haven't added `count` to the list of ref'able builtins. This would only make sense if we were able to **dereference on layer** of the ref without materializing the full extent. I don't think we can do that.